templates/load-message.lua

```diff
@@ -0,0 +1,57 @@
+-- Some parameters to control the behavior. Can be set via Redis keys.
+-- How long (in seconds) should a given buffer bucket be?
+local pBufferBucketInterval = tonumber(redis.call("GET", "conf-bufferBucketInterval")) or 60
+-- How many buffer buckets should we retain?
+local pBufferBucketCount = tonumber(redis.call("GET", "conf-bufferBucketCount")) or 20
+
+-- Derived parameters
+local pBufferLifetime = pBufferBucketInterval * pBufferBucketCount
+
+-- First, parse and restructure the incoming message.
+local message = cjson.decode(ARGV[1])
```

Comment:
Actually, a safer alternative might be to just add a default for `message.app` if it's missing from the payload (and possibly `message.unix_timestamp` as well, although this is added by Logstash itself).

(We are guaranteed that the incoming JSON is actually JSON, since the Redis output drops messages that it can't coerce to JSON: https://github.com/krallin/logstash-output-redis/blob/3fa4b3e0f08dc268a1b3794c6b30b01dd2bc8824/lib/logstash/outputs/redis.rb#L160 — though as you're indicating, we can't know for sure that our tags will be here).